plugins/reports/counter/classes/releases/CounterRelease.inc.php

```diff
@@ -0,0 +1,254 @@
+<?php
+
+/**
+ * @file plugins/generic/sushiLite/classes/releases/CounterRelease.inc.php
+ *
+ * Copyright (c) 2014 University of Pittsburgh
+ * Distributed under the GNU GPL v2 or later. For full terms see the file docs/COPYING.
+ *
+ * @class CounterRelease
+ * @ingroup plugins_reports_counter
+ *
+ * @brief A COUNTER report, base class
+ */
+require_once(dirname(dirname(__FILE__)).'/COUNTER/COUNTER.php');
+
+define('COUNTER_EXCEPTION_WARNING', 0);
+define('COUNTER_EXCEPTION_ERROR', 1);
+define('COUNTER_EXCEPTION_PARTIAL_DATA', 4);
+define('COUNTER_EXCEPTION_NO_DATA', 8);
+define('COUNTER_EXCEPTION_BAD_COLUMNS', 16);
+define('COUNTER_EXCEPTION_BAD_FILTERS', 32);
+define('COUNTER_EXCEPTION_BAD_ORDERBY', 64);
+define('COUNTER_EXCEPTION_BAD_RANGE', 128);
+define('COUNTER_EXCEPTION_INTERNAL', 256);
+
+class CounterRelease {
```

Comment:
Within the context of COUNTER, there is a distinct level of abstraction that happens at the Release level, and another that happens at the Report level.  I had a hard time distinguishing where changes have happened and will happen in the future because I was only really working with Release 4.1, the retracted Release 4.2, and the legacy Release 3.0.

As an example, Release 4.2 was proposed to add a new XML element "ReportAttributes" into the "Report" element.  This would have affected all reports delivered by this COUNTER Release.  I think a set of "Release" classes is needed, even if it's not quite clear yet what will change from release to release.  Thoughts?